live-build/misc/ansible-roles/appliance-build.dcenter/tasks/main.yml

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+- include: ldap.yml
+
+- apt:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - python-pip
+    - libldap2-dev
+    - libsasl2-dev
+    - isc-dhcp-server
+    - bind9
+    - dnsutils
+    - telnet
+  register: result
+  until: result is not failed
+  retries: 3
+  delay: 60
+
+
+- pip:
+    name: pysphere
+    extra_args: --trusted-host devpi.delphix.com -i http://devpi.delphix.com:3141/root/public/+simple/
+    version: 0.1.8
+
+- pip:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - paramiko
```

Comment:
For my own education, is there any particular reason we're installing these dependencies with `pip` rather than the Ubuntu packages `python-paramiko`, `python-ldap`, and `python-requests`? I am not familiar enough with the advantages and disadvantages of each choice to know what is best (especially with regard to upgrade), but I want to make sure we've thought out the long-term consequences of each approach. One disadvantage of the current approach is that the build will fail if PyPi is down (and this has happened before), though this particular disadvantage can be removed by using the abovementioned Artifactory PyPi cache.